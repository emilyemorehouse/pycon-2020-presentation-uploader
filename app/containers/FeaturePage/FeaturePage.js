import React from 'react'
import { Helmet } from 'react-helmet'
import { FormattedMessage } from 'react-intl'

// Components
import { Heading } from 'components/Heading'

// Messages
import messages from './FeaturePage.messages'

// Styles
import { List, ListItem, ListItemTitle } from './FeaturePage.styles'

/*
 * FeaturePage
 *
 * List all the features
 */
export default function FeaturePage() {
  return (
    <div>
      <Helmet>
        <title>Feature Page</title>
        <meta name="description" content="Feature page of React.js Boilerplate application" />
      </Helmet>

      <Heading>
        <FormattedMessage {...messages.header} />
      </Heading>

      <List>
        <ListItem>
          <ListItemTitle>
            <FormattedMessage {...messages.scaffoldingHeader} />
          </ListItemTitle>

          <p>
            <FormattedMessage {...messages.scaffoldingMessage} />
          </p>
        </ListItem>

        <ListItem>
          <ListItemTitle>
            <FormattedMessage {...messages.feedbackHeader} />
          </ListItemTitle>

          <p>
            <FormattedMessage {...messages.feedbackMessage} />
          </p>
        </ListItem>

        <ListItem>
          <ListItemTitle>
            <FormattedMessage {...messages.routingHeader} />
          </ListItemTitle>

          <p>
            <FormattedMessage {...messages.routingMessage} />
          </p>
        </ListItem>

        <ListItem>
          <ListItemTitle>
            <FormattedMessage {...messages.networkHeader} />
          </ListItemTitle>

          <p>
            <FormattedMessage {...messages.networkMessage} />
          </p>
        </ListItem>

        <ListItem>
          <ListItemTitle>
            <FormattedMessage {...messages.intlHeader} />
          </ListItemTitle>

          <p>
            <FormattedMessage {...messages.intlMessage} />
          </p>
        </ListItem>
      </List>
    </div>
  )
}
