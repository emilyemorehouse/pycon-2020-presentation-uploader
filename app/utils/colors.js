const colors = {
  yellow: 'rgb(225, 189, 71);',
  crimson: 'rgb(230,34,45)', // Crimson
  magenta: 'rgb(221,17,117)', // Magenta
  modBlue: 'rgb(65,86,166)', // Mod Blue
  cuttleBlue: 'rgb(24,192,221)', // Cuttle Blue
  darkGrey: 'rgb(68,68,68)', // Dark Grey
  mediumGrey: 'rgb(170,170,170)', // Medium Grey
  lightGrey: 'rgb(238,238,238)', // Light Grey
}

colors.primary = colors.modBlue
colors.secondary = colors.yellow
colors.primaryText = colors.darkGrey
colors.background = colors.lightGrey

export default colors
