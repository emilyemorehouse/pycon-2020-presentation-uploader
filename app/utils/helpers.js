import { isPlainObject } from 'lodash'

const ERROR_DEFAULT = 'An error occurred. Please try again.'
const ERROR_NO_INTERNET = 'Request could not be made. Please check your internet connection.'

export const createWithDoc = ({ envName = '', docFunction = () => {}, component = '' }) => {
  let createComponentWithDoc
  if (envName !== 'production') {
    createComponentWithDoc = docFunction(component) // eslint-disable-line global-require
  }
  return createComponentWithDoc || component
}

export const getErrorMessage = err => {
  // - Axios does not return a response object if the service cannot be reached
  if (!err.response) {
    return ERROR_NO_INTERNET
  }

  // - If we do have a response and data object, use the full error object
  if (err.response && err.response.data) {
    try {
      const errorStringOrObject = Object.values(err.response.data)[0]
      if (isPlainObject(errorStringOrObject)) return errorStringOrObject[0]
      return errorStringOrObject
    } catch (error) {
      // eslint-disable-next-line no-console
      console.warn('Could not infer error message from error response object')
    }
  }

  // If a message has not been set, fallback to our default message
  return ERROR_DEFAULT
}
