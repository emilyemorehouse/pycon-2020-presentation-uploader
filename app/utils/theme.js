import colors from './colors'

const theme = {
  global: {
    font: {
      family: 'Roboto Mono',
      size: '16px',
    },
    colors: {
      brand: colors.primary,
      focus: colors.secondary,
    },
  },
  button: {
    border: {
      radius: '4px',
    },
  },
}

export default theme
