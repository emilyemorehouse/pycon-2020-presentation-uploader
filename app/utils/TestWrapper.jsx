/* eslint-disable react/prop-types */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Grommet } from 'grommet'

import theme from './theme'

export const TestWrapper = ({ children }) => (
  <IntlProvider locale="en">
    <MemoryRouter initialEntries={['/']}>
      <Grommet theme={theme}>{children}</Grommet>
    </MemoryRouter>
  </IntlProvider>
)
