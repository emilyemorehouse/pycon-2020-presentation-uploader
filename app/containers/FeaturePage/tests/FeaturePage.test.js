import React from 'react'
import { render } from '@testing-library/react'

// Components
import { TestWrapper } from 'utils/TestWrapper'
import FeaturePage from '../FeaturePage'

describe('FeaturePage', () => {
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(
      <TestWrapper>
        <FeaturePage />
      </TestWrapper>,
    )

    expect(firstChild).toMatchSnapshot()
  })
})
