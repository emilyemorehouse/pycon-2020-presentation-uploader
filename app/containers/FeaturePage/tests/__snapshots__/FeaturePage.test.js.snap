// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturePage should render its heading 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 blnwXx"
>
  <div>
    <h1
      class="StyledHeading-sc-1rdh4aw-0 kZPGVt"
    >
      <span>
        Features
      </span>
    </h1>
    <ul
      class="FeaturePagestyles__List-sc-1hgx1yj-0 ipghKt"
    >
      <li
        class="FeaturePagestyles__ListItem-sc-1hgx1yj-1 eitokx"
      >
        <p
          class="FeaturePagestyles__ListItemTitle-sc-1hgx1yj-2 fRZFZK"
        >
          <span>
            Quick scaffolding
          </span>
        </p>
        <p>
          <span>
            Automate the creation of components, containers, routes, selectors
  and sagas - and their tests - right from the CLI!
          </span>
        </p>
      </li>
      <li
        class="FeaturePagestyles__ListItem-sc-1hgx1yj-1 eitokx"
      >
        <p
          class="FeaturePagestyles__ListItemTitle-sc-1hgx1yj-2 fRZFZK"
        >
          <span>
            Instant feedback
          </span>
        </p>
        <p>
          <span>
            
      Enjoy the best DX and code your app at the speed of thought! Your
    saved changes to the CSS and JS are reflected instantaneously
    without refreshing the page. Preserve application state even when
    you update something in the underlying code!
    
          </span>
        </p>
      </li>
      <li
        class="FeaturePagestyles__ListItem-sc-1hgx1yj-1 eitokx"
      >
        <p
          class="FeaturePagestyles__ListItemTitle-sc-1hgx1yj-2 fRZFZK"
        >
          <span>
            Industry-standard routing
          </span>
        </p>
        <p>
          <span>
            
      Write composable CSS that's co-located with your components for
    complete modularity. Unique generated class names keep the
    specificity low while eliminating style clashes. Ship only the
    styles that are on the page for the best performance.
    
          </span>
        </p>
      </li>
      <li
        class="FeaturePagestyles__ListItem-sc-1hgx1yj-1 eitokx"
      >
        <p
          class="FeaturePagestyles__ListItemTitle-sc-1hgx1yj-2 fRZFZK"
        >
          <span>
            Offline-first
          </span>
        </p>
        <p>
          <span>
            
      The next frontier in performant web apps: availability without a
      network connection from the instant your users load the app.
    
          </span>
        </p>
      </li>
      <li
        class="FeaturePagestyles__ListItem-sc-1hgx1yj-1 eitokx"
      >
        <p
          class="FeaturePagestyles__ListItemTitle-sc-1hgx1yj-2 fRZFZK"
        >
          <span>
            Complete i18n Standard Internationalization & Pluralization
          </span>
        </p>
        <p>
          <span>
            Scalable apps need to support multiple languages, easily add and support multiple languages with \`react-intl\`.
          </span>
        </p>
      </li>
    </ul>
  </div>
</div>
`;
